CONTRIBUTING.md

```diff
@@ -70,6 +70,35 @@ issues][open-issues].
 Remember to always include (when applicable): i) unit tests which exercises the new code,
 ii) documentation, iii) a note in the [CHANGELOG.md](CHANGELOG.md) file.
 
+### Code formatting
+
+Ferrite uses [Runic.jl][runic] for code formatting. You can install Runic with the following
+command:
+
+```shell
+julia --project=@runic -e 'using Pkg; Pkg.add("Runic")'
+```
+
+and you can then run the formatter from the root of the repository with:
+
+```shell
+julia --project=@runic -e 'using Runic; exit(Runic.main(ARGS))' -- --inplace .
+```
+
+Refer to the Runic.jl [README][runic-readme] for more details (such as editor integration).
+
+
+### pre-commit hooks
+
+Ferrite uses [pre-commit] to run various checks (including Runic formatting). After
+installing pre-commit (refer to the [pre-commit installation
+instructions][pre-commit-install]) run the following from the root of the repository to
+configure pre-commit to run before each commit:
+
+```shell
+pre-commit install
+```
+
```

Comment:
Not sure if this is completely accurate, but I assume that it is at least a required step? I installed first anyway, and didn't test without installing first...

```suggestion
Note that this requires `Runic.jl` to be installed in the shared `@runic` environment according to the instructions above
```